### PR TITLE
APM: Update status templates to show probabilistic sampler when enabled

### PR DESCRIPTION
**Pull request by ajgajg1134**
### What does this PR do?
Show the probabilistic sampler when enabled, this also hides the "head based sampling rates" when enabled as those priorities are ignored when the probabilistic sampler is enabled.

### Motivation
This will help customers and support debug their setup and reduce confusion on their configuration

### Describe how to test/QA your changes
Added a unit test here to verify config output as well as manually tested all three possible status templates locally.
`agent status`
`agent launch-gui`
`trace-agent info`

### CODE REVIEW

File: releasenotes/notes/probabilistic-sampler-status-2ac8b77c4f96bfae.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+fixes:
+  - |
+    APM: Show probabilistic sampling configuration in agent status when enabled.
```
Comment (brett0000FF) [inline releasenotes/notes/probabilistic-sampler-status-2ac8b77c4f96bfae.yaml:11]:
```suggestion
    APM: Show probabilistic sampling configuration in Agent status when enabled.
```

**Review by brett0000FF — APPROVED** [review]
Approved for release note with minor suggestion.

File: pkg/trace/info/info_test.go [hunk 2]
```diff
@@ -177,6 +178,38 @@ func TestInfo(t *testing.T) {
...
+
+	var buf bytes.Buffer
+	err = Info(&buf, conf)
+	assert.NoError(err)
+	info := buf.String()
+	assert.NotEmpty(info)
+	t.Logf("Info:\n%s\n", info)
+	expectedInfo, err := os.ReadFile("./testdata/psp.info")
+	re := regexp.MustCompile(`\r\n`)
+	expectedInfoString := re.ReplaceAllString(string(expectedInfo), "\n")
```
Comment (ichinaski) [inline pkg/trace/info/info_test.go:208]: I'm likely missing something, but why don't we just ensure `psp.info` contains `\n` line breaks?

**Review by ichinaski — APPROVED** [review]
LGTM, approved with a small question.

File: pkg/trace/info/info_test.go (hunk 2 above)
Comment (ajgajg1134) [reply pkg/trace/info/info_test.go:208]: I _think_ this is to deal with inconsistencies running this test on Windows? (to be honest here I copied this logic from a different test and didn't think super critically about this line)

**Pull request merged**